chapter15/dtj/ex4

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use Scalar::Util qw(looks_like_number);
+use 5.010001;
+
+chomp ( my $number = <STDIN>);
+my @empty_array;
+my @divisors;
+
+given ($number) {
+  # looks_like_number might be cheating here, but I don't know how to do it with smart matching (unless it's with a regex....)
+  when ( !looks_like_number( $number ) ) { die "That's not a number\n" }
```

Comment:
I think you could use the regex /^\d+$/ to check it's a number with smart matching (since we only want to check it's an integer)